napari/layers/shapes/_shapes_utils.py

```diff
@@ -33,10 +33,15 @@ def _is_convex(poly: npt.NDArray) -> bool:
     bool
         True if the given polygon is convex.
     """
-    fst = poly
-    snd = np.roll(poly, -1, axis=0)
-    thrd = np.roll(poly, -2, axis=0)
-    return np.unique(orientation(fst.T, snd.T, thrd.T)).size == 1
+    fst = poly[:-2]
+    snd = poly[1:-1]
+    thrd = poly[2:]
+    orn_set = np.unique(orientation(fst.T, snd.T, thrd.T))
+    if orn_set.size != 1:
+        return False
+    return orn_set[0] == orientation(poly[-2], poly[-1], poly[0]) and orn_set[
+        0
+    ] == orientation(poly[-1], poly[0], poly[1])
```

Comment:
Added parentheses so the formatter does not puke like that :P

```suggestion
    return (orn_set[0] == orientation(poly[-2], poly[-1], poly[0])) and (
        orn_set[0] == orientation(poly[-1], poly[0], poly[1])
    )
```

I don't see how this code is equivalent, but I'm guessing there's tests for this to make sure things are not changed?

Reply:
You do not see why my code is equivalent?